communication/src/networking.rs

```diff
@@ -109,8 +113,13 @@ pub fn await_connections(addresses: Arc<Vec<String>>, my_index: usize, noisy: bo
     for _ in (my_index + 1) .. addresses.len() {
         let mut stream = listener.accept()?.0;
         stream.set_nodelay(true).expect("set_nodelay call failed");
-        let mut buffer = [0u8;8];
-        stream.read_exact(&mut buffer).expect("failed to read worker index");
+        let mut buffer = [0u8;16];
+        stream.read_exact(&mut buffer)?;
+        let (magic, mut buffer) = unsafe { decode::<u64>(&mut buffer) }.expect("failed to decode magic");
+        if magic != &HANDSHAKE_MAGIC {
+            return Err(io::Error::new(io::ErrorKind::InvalidData,
+                "received incorrect timely handshake"));
+        }
         let identifier = unsafe { decode::<u64>(&mut buffer) }.expect("failed to decode worker index").0.clone() as usize;
```

Comment:
This isn't going to read out the correct identifier, will it?  It seems like it will just read out the magic number again.

Reply:
No, it works! `buffer` is advanced (i.e., reassigned to the remaining bytes that abomonation returns) on L118.

Reply:
You are entirely correct! Something feels icky, but we can tidy that up in post. :D